pkg/install/1-installresources_test.go

```diff
@@ -0,0 +1,70 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"testing"
+
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	aztypes "github.com/openshift/installer/pkg/types/azure"
+)
+
+func TestZones(t *testing.T) {
+	tests := []struct {
+		zones      []string
+		replicas   *int64
+		shouldPass bool
+	}{
+		{
+			zones:      []string{"1"},
+			replicas:   int64pointer(1),
+			shouldPass: true,
+		},
+		{
+			zones:      []string{"1", "2"},
+			replicas:   int64pointer(2),
+			shouldPass: true,
+		},
+		{
+			zones:      []string{"1", "2"},
+			replicas:   int64pointer(1),
+			shouldPass: false,
+		},
+		{
+			zones:      []string{"1"},
+			replicas:   int64pointer(2),
+			shouldPass: true,
+		},
+	}
+
+	for _, tt := range tests {
+		_, err := zones(&installconfig.InstallConfig{
+			Config: &types.InstallConfig{
+				ControlPlane: &types.MachinePool{
+					Platform: types.MachinePoolPlatform{
+						Azure: &aztypes.MachinePool{
+							Zones: tt.zones,
+						},
+					},
+					Replicas: tt.replicas,
+				},
+			},
+		})
+		if tt.shouldPass {
+			if err != nil {
+				t.Error(err)
+			}
+		} else {
+			if err == nil {
+				t.Errorf("should fail with %d zones and %d replicas", len(tt.zones), *tt.replicas)
+			}
+		}
+	}
+}
+
+func int64pointer(i int) *int64 {
+	i64 := int64(i)
+	return &i64
+}
```

Comment:
write `to.Int64Ptr()` - grep for other usage

Reply:
I forgot that existed :(